direction_matrix.go

```diff
@@ -0,0 +1,59 @@
+package mapbox
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+const (
+	directionsMatrix = "directions-matrix"
+)
+
+type DirectionsMatrixRequest struct {
+	// required
+	Profile Profile
+	Coordinates Coordinates
+
+	// optional
+	Annotations Annotations
+	Approaches Approaches
+	Destinations Destinations
+	Sources Sources
+	FallbackSpeed FallbackSpeed
+
+}
+
+type DirectionsMatrixResponse struct {
+	Code string `json:"code"`
+	Durations [][]float64 `json:"durations"`
+	Distances [][]float64 `json:"distances"`
+	Destinations []Waypoint `json:"destinations"`
+	Sources []Waypoint `json:"sources"`
+}
+
+// https://docs.mapbox.com/api/navigation/#matrix
+func (c *Client) DirectionsMatrix(ctx context.Context, req DirectionsMatrixRequest) (*DirectionsMatrixResponse, error) {
```

Comment:
This is a little weird having the method of client not in the file with client.

Maybe have them in the same file, or something like this: ?

client.go
```go
func (c *Client) DirectionsMatrix(ctx context.Context, req DirectionsMatrixRequest) (*DirectionsMatrixResponse, error) {
    return directionsMatrix(ctx, c, req)
}
```

directions_matrix.go
```go
func directionsMatrix(ctx context.Context, c *Client, req DirectionsMatrixRequest) (*DirectionsMatrixResponse, error) {
    /// the implementation you have here
    /// .....
}
```